command/format/state.go

```diff
@@ -98,78 +98,109 @@ func formatStateModule(p blockBodyDiffPrinter, m *states.Module, schemas *terraf
 	// Go through each resource and begin building up the output.
 	for _, key := range names {
 		for k, v := range m.Resources[key].Instances {
+			// keep these in order to keep the current object first, and
+			// provide deterministic output for the deposed objects
+			type obj struct {
+				header   string
+				instance *states.ResourceInstanceObjectSrc
+			}
+			instances := []obj{}
+
 			addr := m.Resources[key].Addr
 
 			taintStr := ""
-			if v.Current.Status == 'T' {
+			if v.Current != nil && v.Current.Status == 'T' {
 				taintStr = " (tainted)"
 			}
-			p.buf.WriteString(fmt.Sprintf("# %s:%s\n", addr.Absolute(m.Addr).Instance(k), taintStr))
-
-			var schema *configschema.Block
-			provider := m.Resources[key].ProviderConfig.ProviderConfig.StringCompact()
-			if _, exists := schemas.Providers[provider]; !exists {
-				// This should never happen in normal use because we should've
-				// loaded all of the schemas and checked things prior to this
-				// point. We can't return errors here, but since this is UI code
-				// we will try to do _something_ reasonable.
-				p.buf.WriteString(fmt.Sprintf("# missing schema for provider %q\n\n", provider))
-				continue
+
+			instances = append(instances,
+				obj{fmt.Sprintf("# %s:%s\n", addr.Absolute(m.Addr).Instance(k), taintStr), v.Current})
+
+			for dk, v := range v.Deposed {
+				instances = append(instances,
+					obj{fmt.Sprintf("# %s: (deposed object %s)\n", addr.Absolute(m.Addr).Instance(k), dk), v})
 			}
 
-			switch addr.Mode {
-			case addrs.ManagedResourceMode:
-				schema, _ = schemas.ResourceTypeConfig(
-					provider,
-					addr.Mode,
-					addr.Type,
-				)
-				if schema == nil {
-					p.buf.WriteString(fmt.Sprintf(
-						"# missing schema for provider %q resource type %s\n\n", provider, addr.Type))
+			// sort the deposed objects
+			sort.Slice(instances[1:], func(i, j int) bool {
+				return instances[i+1].header < instances[j+1].header
+			})
+
+			for _, obj := range instances {
+				header := obj.header
+				instance := obj.instance
+				p.buf.WriteString(header)
+				if instance == nil {
```

Comment:
Is this the only usage of `instance`? If so why not use `obj.instance`?

Reply:
Ah, `instance` is created because it's in `instance.Decode` further down.

Reply:
Cool, missed it in the diff, thanks!